lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -242,6 +247,9 @@ data ByronWalletStyle
     | Icarus
     | Trezor
     | Ledger
+    deriving (Show, Generic, Eq, Bounded, Enum)
+    deriving (ToText, FromText)
+            via (BoundedEnum 'KebabLowerCase ByronWalletStyle)
```

Comment:
This seems like an odd way of writing: 

```hs
instance FromText ByronWalletStyle where
    fromText = fromTextToBoundedEnum SnakeLowerCase

instance ToText ByronWalletStyle where
    toText = toTextFromBoundedEnum SnakeLowerCase
```

In order to save 2 lines, we have to introduce `DerivingVia` + a whole new type `BoundedEnum`. Please don't.
To be clear, I am not advocating against `DerivingVia` but this is not the right place for introducing this because:

a) We really aren't saving much here... The instances are already straightforward.
b) Rest of the code using the casing functions don't use DerivingVia. So, if something like this is to be introduced, then it needs to be done consistently across all call-site where `SnakeLowerCase` are used. Else, we now have 2 different ways of maintaining parts of code that are doing the exact same thing. 
c) This PR isn't the right place for going through b).

Reply:
> We really aren't saving much here... The instances are already straightforward.

Personally I find:
- It's easier to see that the `instance` declarations belong to `ByronWalletStyle` as a `deriving` statement directly under the type declaration. Multiple instances following the type declaration make it more difficult to see the structure in the module.
- `DerivingVia` guarantees that `ToText` and `FromText` are consistently derived using the same mechanism. I.e both via `BoundedEnum` and both using `KebabLowerCase`.

> then it needs to be done consistently across all call-site where

I thought some diversity / incremental-ity could be fine, but sure.

>  but this is not the right place for introducing this because:

Fair point.

Reply:
I agree with both of your first point and again, I am not against using DerivingVia. But, if we use `A`, we do it consistently. Merging this as it is would introduce `B` and keep `A` in all other places. This makes the code less maintainable. The key for maintainability is consistency. If patterns are not obvious, then we will inevitably deviate from existing patterns and soon, there will be a `C` way of doing it.